### PR TITLE
Cleanup and enhance misc reflection/attribute tests

### PR DESCRIPTION
**Pull request by hughbe**

### CODE REVIEW

File: src/System.Runtime/tests/System/Reflection/AssemblyFlagsAttributeTests.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    {
+        [Theory]
+        [InlineData(AssemblyNameFlags.None, 0)]
+        [InlineData(AssemblyNameFlags.EnableJITcompileOptimizer, 16384)]
+        [InlineData(AssemblyNameFlags.EnableJITcompileTracking, 32768)]
+        [InlineData(AssemblyNameFlags.PublicKey, 1)]
+        [InlineData(AssemblyNameFlags.Retargetable, 256)]
+        [InlineData(int.MinValue, 2147483648)]
+        [InlineData((AssemblyNameFlags)(AssemblyNameFlags.None - 1), uint.MaxValue)]
+        [InlineData((AssemblyNameFlags)int.MaxValue, 2147483647)]
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/Reflection/AssemblyFlagsAttributeTests.cs]: Why pass in both assemblyFlags and expectedFlags?  Flags and AssemblyFlags both return the AssemblyNameFlags, just cast to uint and int, respectively.  We shouldn't need to hardcode the integral value into the test like this, nor duplicate it.

File: src/System.Runtime/tests/System/Reflection/AssemblyFlagsAttributeTests.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        }
+
+        [Theory]
+        [InlineData(0, 0)]
+        [InlineData(16384, 16384)]
+        [InlineData(32768, 32768)]
+        [InlineData(1, 1)]
+        [InlineData(256, 256)]
+        [InlineData(uint.MaxValue, -1)]
+        public void Ctor_UInt(uint flags, int expectedAssemblyFlags)
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/Reflection/AssemblyFlagsAttributeTests.cs]: Ditto, here and in the other tests as well.

File: src/System.Runtime/tests/System/Reflection/AssemblyInformationalVersionAttributeTests.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+using Xunit;
+
+namespace System.Reflection.Tests
+{
+    public class AssemblyInformationalVersionAttributeTests
+    {
+        [Theory]
+        [InlineData(null)]
+        [InlineData("")]
+        [InlineData("delaySign")]
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/Reflection/AssemblyInformationalVersionAttributeTests.cs]: "delaySign"?

File: src/System.Runtime/tests/System/Reflection/AssemblyProductAttributeTests.cs [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+using Xunit;
+
+namespace System.Reflection.Tests
+{
+    public class AssemblyProductAttributeTests
+    {
+        [Theory]
+        [InlineData(null)]
+        [InlineData("")]
+        [InlineData("product")]
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/Reflection/AssemblyProductAttributeTests.cs]: Nit: here and in some of the other tests, can you use values that are more likely to actually be used?  For example, the original test has ".NET Core" here... that'd be better than "product".  As another example, the examples earlier that were for versions in the original tests actually used version numbers, but these tests just use "version".  If you want to also use "version", that's fine, but it'd be good to have inputs that exemplify canonical usage.

File: src/System.Runtime/tests/System/Reflection/AssemblyFlagsAttributeTests.cs (hunk 1 above)
Comment (hughbe) [reply src/System.Runtime/tests/System/Reflection/AssemblyFlagsAttributeTests.cs]: I didn't want to simply copy the src implementation in the tests, but i fixed

Comment (stephentoub) [reply src/System.Runtime/tests/System/Reflection/AssemblyFlagsAttributeTests.cs]: Thanks. Hardcoding the value was essentially copying the implementation, too... there's not much more you can do when the entire implementation is a simple cast.

**Pull request merged**